doc/source/tune.rst

```diff
@@ -0,0 +1,160 @@
+
+
+Getting Started
+--------------- 
+
+::
+
+    from ray.tune import register_trainable, grid_search, run_experiments
+
+    def my_func(config, reporter):
+        import time, numpy as np
+        i = 0
+        while True:
+            reporter(timesteps_total=i, mean_accuracy=i ** config["alpha"])
+            i += config["beta"]
+            time.sleep(.01)
+
+    register_trainable("my_func", my_func)
+
+    run_experiments({
+        "my_experiment": {
+            "run": "my_func",
+            "resources": { "cpu": 1, "gpu": 0 },
+            "stop": { "mean_accuracy": 100 },
+            "config": {
+                "alpha": grid_search([0.2, 0.4, 0.6]),
+                "beta": grid_search([1, 2]),
+            },
+        }
+    })
+
+
+This script runs a small grid search over the ``my_func`` function using ray.tune, reporting status on the command line until the stopping condition of ``mean_accuracy >= 100`` is reached:
+
+::
+
+    == Status ==
+    Using FIFO scheduling algorithm.
+    Resources used: 4/8 CPUs, 0/0 GPUs
+    Tensorboard logdir: /tmp/ray/my_experiment
+     - my_func_0_alpha=0.2,beta=1:	RUNNING [pid=6778], 209 s, 20604 ts, 7.29 acc
+     - my_func_1_alpha=0.4,beta=1:	RUNNING [pid=6780], 208 s, 20522 ts, 53.1 acc
+     - my_func_2_alpha=0.6,beta=1:	TERMINATED [pid=6789], 21 s, 2190 ts, 101 acc
+     - my_func_3_alpha=0.2,beta=2:	RUNNING [pid=6791], 208 s, 41004 ts, 8.37 acc
+     - my_func_4_alpha=0.4,beta=2:	RUNNING [pid=6800], 209 s, 41204 ts, 70.1 acc
+     - my_func_5_alpha=0.6,beta=2:	TERMINATED [pid=6809], 10 s, 2164 ts, 100 acc
+
+In order to report incremental progress, ``my_func`` periodically calls the ``reporter`` function passed in by Ray.tune to return the current timestep and other metrics as defined in `ray.tune.result.TrainingResult <https://github.com/ray-project/ray/blob/master/python/ray/tune/result.py>`__.
+
+Visualizing Results
+-------------------
+
+Ray.tune logs trial results to a unique directory per experiment, e.g. ``/tmp/ray/my_experiment`` in the above example. The log records are compatible with a number of visualization tools:
+
+To visualize learning in tensorboard, run:
+
+::
+
+    $ pip install tensorboard
+    $ tensorboard --logdir=/tmp/ray/my_experiment
+
+.. image:: ray-tune-tensorboard.png
+
+To use rllab's VisKit (you may have to install some dependencies), run:
+
+::
+
+    $ git clone https://github.com/rll/rllab.git
+    $ python rllab/rllab/viskit/frontend.py /tmp/ray/my_experiment
+
+.. image:: ray-tune-viskit.png
+
+Finally, to view the results with a `parallel coordinates visualization <https://en.wikipedia.org/wiki/Parallel_coordinates>`__, open `ParalleCoordinatesVisualization.ipynb <https://github.com/ray-project/ray/blob/master/python/ray/tune/ParallelCoordinatesVisualization.ipynb>`__ as follows and run its cells:
+
+::
+
+    $ cd $RAY_HOME/python/ray/tune
+    $ jupyter-notebook ParallelCoordinatesVisualization.ipynb
```

Comment:
Can we also add an image for this?

Reply:
I tried, it looks not so good so I left it out.